hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/HoodieCompactor.java

```diff
@@ -261,6 +287,50 @@ public List<WriteStatus> compact(HoodieCompactionHandler compactionHandler,
     }).collect(toList());
   }
 
+  /**
+   * Execute a single compaction operation and report back status.
+   */
+  public List<WriteStatus> compact(HoodieCompactionHandler compactionHandler,
+                                   HoodieTableMetaClient metaClient,
+                                   HoodieWriteConfig config,
+                                   CompactionOperation operation,
```

Comment:
looks like lot of the args are not required here for FG based compaction. 
can you clean them up.

Reply:
All of these are required.